Backpack/src/main/java/net/skyscanner/backpack/map/BpkMapMarker.kt

```diff
@@ -0,0 +1,155 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018-2021 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.map
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.view.View
+import androidx.annotation.DrawableRes
+import androidx.appcompat.content.res.AppCompatResources
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.model.BitmapDescriptor
+import com.google.android.gms.maps.model.BitmapDescriptorFactory
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.Marker
+import com.google.android.gms.maps.model.MarkerOptions
+import net.skyscanner.backpack.R
+import net.skyscanner.backpack.map.view.createBpkMarkerView
+import net.skyscanner.backpack.util.rasterize
+
+/**
+ * An icon which will be displayed on the badge.
+ */
+var Marker.icon: Int
```

Comment:
Google Maps doesn't have a way to put an icon in marker. You can, however, associate it with some tag (an object).

We use this object to store any custom-related properties. This is extension property - synthetic  way to expand the class functionality if you don't have access to its source code editing. Basically it's a syntax sugar for calling utility functions.
 
https://kotlinlang.org/docs/extensions.html